### PR TITLE
Fixes #24756: Accepting one selected pending node pops up a windows that accepts them all

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/24756

Some code was unused and I prefer to clean it up, dead code in the accept snippet is a stick to get cyber-hit.

The correction is trivial and just filtre the list of nodes with the list of select ones.

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/node/AcceptNode.scala [hunk 1]
```diff
@@ -79,20 +76,8 @@ class AcceptNode extends Loggable {
   val historyRepos     = RudderConfig.inventoryHistoryJdbcRepository
   val logRepository    = RudderConfig.eventLogRepository
   val acceptedNodesDit = RudderConfig.acceptedNodesDit
-  val pendingNodeDit   = RudderConfig.pendingNodesDit
```
Comment (fanf) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/node/AcceptNode.scala:82]: dead code

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/node/AcceptNode.scala [hunk 2]
```diff
@@ -127,21 +112,6 @@ class AcceptNode extends Loggable {
...
-  /**
-   * Retrieve the last inventory for the selected server
-   */
-  def retrieveLastVersions(uuid: NodeId): Option[DateTime] = {
-    historyRepos.versions(uuid).toBox match {
-      case Full(list) if (list.nonEmpty) => Some(list.head)
-      case _                             => None
-    }
-  }
-
```
Comment (fanf) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/node/AcceptNode.scala:144]: dead code

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/node/AcceptNode.scala [hunk 3]
```diff
@@ -264,7 +234,7 @@ class AcceptNode extends Loggable {
 
     nodeFactRepository
       .getAll()(QueryContext.todoQC, SelectNodeStatus.Pending)
-      .map(_.values)
+      .map(_.collect { case (id, n) if listNode.contains(id) => n })
```
Comment (fanf) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/node/AcceptNode.scala:237]: the actual correction for the main problem.

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/node/AcceptNode.scala [hunk 4]
```diff
@@ -341,7 +311,7 @@ class AcceptNode extends Loggable {
           (selectAll, { e => <input type="checkbox" name="serverids" value={e.id.value.toString}/> })
         ),
         """,{ "sWidth": "10%" },{ "sWidth": "11%", "bSortable":false },{ "sWidth": "2%", "bSortable":false }""",
-        true
+        paginate = true
```
Comment (fanf) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/node/AcceptNode.scala:314]: scala 3 compliance

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/node/AcceptNode.scala [hunk 5]
```diff
@@ -174,7 +144,7 @@ class AcceptNode extends Loggable {
         case e: EmptyBox =>
           logger.error(s"Add new node '$id.value' lead to Failure.", e)
           S.error(<span class="error">Error while accepting node(s).</span>)
-        case Full(inventory) =>
+        case Full(_) =>
```
Comment (fanf) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/node/AcceptNode.scala:147]: scala 3 compliance (unused var should be unamed)

**Review by clarktsiory — APPROVED** [review]
LGTM

**Pull request merged**